lib/nautic_net/data_set_recorder.ex

```diff
@@ -102,4 +100,20 @@ defmodule NauticNet.DataSetRecorder do
   def terminate(_reason, state) do
     save_data_points(state.data_points, state)
   end
+
+  @doc """
+  Returns the directory to save the datasets to. If the directory does not exist
+  create it before returning
+  """
+  @spec dataset_directory(Keyword.t()) :: String.t()
+  def dataset_directory(opts) do
+    if tmp_dir = opts[:temp_dir] do
+      :ok = File.mkdir_p!(tmp_dir)
+      tmp_dir
+    else
+      dir = Application.get_env(:nautic_net_device, :data_set_directory, Path.join(System.tmp_dir!(), "datasets"))
```

Comment:
@schrockwell Good catch on this. I have updated to be runtime always now on host and when on target then it will be hardcoded to `/data/datasets`